src/main/java/pl/javastart/task/Truck.java

```diff
@@ -0,0 +1,26 @@
+package pl.javastart.task;
+
+public class Truck extends Car {
+    private double loadWeight;
+
+    public double getLoadWeight() {
+        return loadWeight;
+    }
+
+    public void setLoadWeight(double loadWeight) {
+        this.loadWeight = loadWeight;
+    }
+
+    public Truck(String name, double tankCapacity, double averageConsumption, boolean isClimaTurnedOn, double loadWeight) {
+        super(name, tankCapacity, averageConsumption, isClimaTurnedOn);
+        this.loadWeight = loadWeight;
+    }
+
+    public double countRange(Vehicle vehicle, boolean isClimaTurnedOn, double loadWeight) {
+        if (isClimaTurnedOn) {
+            return ((vehicle.getTankCapacity() / (vehicle.getAverageConsumption() + 1.6 + 0.005 * loadWeight) * 100));
```

Comment:
Unikaj stosowania tzw. "magicznych liczb". Zastąp ją dobrze nazwaną stałą https://youtu.be/eY1li8xwX3M